graphql/subcourse/fields.ts

```diff
@@ -79,9 +79,21 @@ export class ExtendedFieldsSubcourseResolver {
     }
 
     @FieldResolver((returns) => [Pupil])
-    @Authorized(Role.ADMIN)
+    @Authorized(Role.ADMIN, Role.INSTRUCTOR)
     @LimitEstimated(100)
-    async participants(@Root() subcourse: Subcourse) {
+    async participants(@Ctx() context: GraphQLContext, @Root() subcourse: Subcourse) {
+        if (!isElevated(context)) {
+            return await prisma.pupil.findMany({
+                select: { firstname: true, lastname: true, grade: true },
```

Comment:
I initially thought about something like Role.IS_INSTRUCTOR_OF analogous to Role.OWNER, but the more I think about it that would be quite complex to implement (securely). I actually like the idea of only returning the fields here, as it can't go wrong. I would however like to distinguish the resolvers then, as `courses { participants { email }}` might now suddenly return null values (and further field resolvers might fail in weird ways). I would prefer if we have two resolvers instead, `participantsAsPupil` which is only accessible to Role.ADMIN and which returns regular pupils, and `participants` which is accessible to Role.OWNER, and which only returns the (firstname, lastname, grade). For the `participants` resolver I guess we should then also introduce a new type (`@ObjectType() class Participant { ... }`) so that the participants resolver only has the fields that are actually available.

Reply:
And I guess the participants resolver should also return the pupil id, so that the instructor can then call mutations for specific participants (i.e. send them a message or the participation certificate) ... Sorry for coming up with all this now, just didn't think this through when writing the issue ...

Reply:
No problemo, have implemented it now. Couldnt test the participant resolver for now, because I am not sure how to preserve the login token from legacyLogin in Apollo explorer. But is this what you imagined?

Reply:
Yes, very nice. Though the diff looks weird, maybe merge master into the branch before squash merging (not sure what would happen when squash merging in the current state)...

Reply:
Ok I already merged (rebased) master into the branch, but maybe not entirely? I will take a look at it again later

Reply:
Ah, seems this is by design (https://stackoverflow.com/questions/63603247/github-showing-an-incorrect-diff-between-main-branch-and-pr-branch), so merging (git merge master) works better than rebasing with Github PRs. Anyways, merging fixed it and the diff is now correct :)

Reply:
Ah ok I see :+1: